src/diffusers/utils/import_utils.py

```diff
@@ -295,6 +295,20 @@
 except importlib_metadata.PackageNotFoundError:
     _torchvision_available = False
 
+# Taken from https://stackoverflow.com/a/39662359.
+# Adapted to make it work with Google colab as well.
+_is_notebook = False
+try:
+    shell_class = get_ipython().__class__  # type: ignore # noqa: F821
+    for parent_class in shell_class.__mro__:  # e.g. "is subclass of"
+        if parent_class.__name__ == "ZMQInteractiveShell":
```

Comment:
Is this universal enough?

Reply:
I also forgot to mention this too. Taken from [`huggingface_hub`](https://github.com/huggingface/huggingface_hub/blob/57ee5fc97c20b884d97bac5e589204f6c4fb8c1b/src/huggingface_hub/utils/_runtime.py#L251-L264).